### PR TITLE
Refactor context lookup logic yet again

### LINKED ISSUE #3054: Mac and Windows activation time is higher in 1.14.0
(Title)

Both Mac and WIndows activation time spiked significantly in 1.14.0. Doesn't seem to have occurred in Linux.

In the image, yellow line is Windows activation, purple is Mac.

image

### PR DESCRIPTION
**Pull request by bwateratmsft**
Partially fixes #3054; in most cases on Docker Desktop (Mac and Windows) we will be able to use the gRPC daemon for the context lookup which is quite a bit faster than the CLI.

### CODE REVIEW

File: src/docker/ContextManager.ts [hunk 1]
```diff
@@ -195,133 +198,200 @@ export class DockerContextManager implements ContextManager, Disposable {
...
                 DockerEndpoint: isWindows() ? WindowsLocalPipe : UnixLocalPipe,
-            } as DockerContext];
+            } as DockerContext;
         }
 
-        return loadResult;
+        return undefined;
+    }
+
+    private async tryGetContextsFromGrpc(actionContext: IActionContext, maybeFixedContextName: string | undefined): Promise<DockerContext[] | undefined> {
```
Comment (karolz-ms) [inline src/docker/ContextManager.ts:310]: gRPC is an implementation detail. Suggest `tryGetContextsUsingAPI`, or something along these lines

**Review by karolz-ms — COMMENTED** [review]
Looks great overall, just wondering about `async` usage

File: src/docker/ContextManager.ts [hunk 2]
```diff
@@ -195,133 +198,200 @@ export class DockerContextManager implements ContextManager, Disposable {
...
-                    // Blank lines should be skipped
-                    if (!line) {
-                        continue;
-                    }
-
-                    const context = JSON.parse(line) as DockerContext;
-                    result.push(toDockerContext(context));
-                }
-            }
+    private async tryGetContextFromSettings(actionContext: IActionContext): Promise<DockerContext | undefined | string> {
```
[the remark this answers is not in the document]
Comment (bwateratmsft) [reply src/docker/ContextManager.ts:251]: Good point. I don't think any of these three shortcut ones need to be async.

File: src/docker/ContextManager.ts (hunk 1 above)
Comment (bwateratmsft) [reply src/docker/ContextManager.ts:310]: Good idea.

**Review by karolz-ms — APPROVED** [review]

File: src/docker/DockerServeClient/DockerServeClient.ts [hunk 3]
```diff
@@ -212,6 +219,29 @@ export class DockerServeClient extends ContextChangeCancelClient implements Dock
...
+                Name: ctx.name,
+                ContextType: ctx.contexttype as ContextType,
+                Current: ctx.current,
+                DockerEndpoint: ctx.dockerEndpoint?.host,
+                Description: ctx.description,
+                CreatedTime: undefined,
+            };
+        });
+
+        // Workaround for <INSERT BUG LINK>: if no context is marked as Current=true, that means the environment has a fixed context or otherwise the default context is selected, so overwrite that property
```
Comment (bwateratmsft) [inline src/docker/DockerServeClient/DockerServeClient.ts:237]: Insert said bug link

**Pull request merged**